chore(deps): Upgrade tailwindcss (v1.9 to v3), autoprefixer, & all SB deps, replace postcss deps, remove other deps

LZoog opened the pull request:
Because:
* We want to upgrade to Tailwind v3 and update other related outdated dependencies

This commit:
* Upgrades storybook packages across the board, upgrades autoprefixer, removes postcss-7 compatibility package and postcss-cli, installs postcss per Tailwind requirement
* Upgrades fxa-admin-panel to Tailwind v3
* Upgrades fxa-react to Tailwind v3, changing Portal from SCSS to CSS to remove a build step
* Upgrades fxa-settings to Tailwind v3, first transforming our SCSS to CSS before it’s sent to PostCSS. There will be a follow up to change our SCSS files to CSS and use PostCSS plugins instead
* Upgrades fxa-payments-server to Tailwind v3, decoupling SCSS from the Tailwind out file since payments will be using Tailwind exclusively in the next few months. Makes Storybook more consistent with other packages

---

Going to leave everything below, it's kind of a repeat of what I have in the commit message but w/e.

**Follow up issues filed**: #13477, #13478, #13479

## helpful links when reviewing
* https://v2.tailwindcss.com/docs/upgrading-to-v2
* https://tailwindcss.com/docs/upgrade-guide (v2 to v3)
* https://storybook.js.org/addons/@storybook/addon-postcss
* Replacing SCSS: 
  * https://tailwindcss.com/docs/using-with-preprocessors
  * https://github.com/postcss/postcss-nested
  * https://github.com/postcss/postcss-import

## what this PR does besides upgrade stuff

We can remove SCSS dependence later in favor of postcss-nested and other postcss plugins which will remove one step from our build pipeline for settings. I went ahead and removed it where I could.

### fxa-admin-panel
* not much needed changing here since admin-panel wasn't using SCSS

### fxa-react
* Portal was the only component using SCSS. To remove a build step here, I went ahead and converted it to CSS.

### fxa-settings
* Tailwind v3 requires us to transform our SCSS into CSS before it's sent into PostCSS. I can file follow a follow up to change our SCSS files to CSS and use PostCSS plugins (I believe just `postcss-nested` and `postcss-import`) instead to remove a build step.

### fxa-payments-server
* I decoupled Tailwind and the rest of payments' SCSS here. At the time of writing the `build-css` step handles just the Tailwind command since all components are importing their SCSS directly anyway. I experimented with a couple of different approaches and this one seemed like the cleanest considering we're going to be removing all the SCSS here anyway when payments is converted to Tailwind soon.

fixes #12550
fixes #12154
fixes #12154

LZoog commented on packages/fxa-content-server/app/styles/_variables.scss:
```diff
@@ -1,4 +1,4 @@
-@import '~photon-colors/colors.scss';
+@import '../../node_modules/photon-colors/colors.scss';
```
will revert this back, had to do this for a payments approach I didn't go with

pdehaan commented on packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/index.scss:
```diff
@@ -92,7 +92,7 @@
 
     &.new-plan-label {
-      background-image: url('../../../images/down-arrow.png');
+      // background-image: url('../../../images/down-arrow.png');
```
Q: why remove this?

LZoog replied:
This was prior to me addressing getting this image loading in Storybook, it needed to use `file-loader`.

pdehaan commented on packages/fxa-settings/src/components/GetDataTrio/index.tsx:
```diff
@@ -102,7 +102,7 @@ export const GetDataTrio = ({ value, onAction }: GetDataTrioProps) => {
           }}
           data-testid="databutton-print"
-          className="w-12 h-12 relative inline-block text-grey-500 rounded active:text-blue-500 active:outline-black-dotted focus:outline-black-dotted hover:bg-grey-50"
+          className="w-12 h-12 relative inline-block text-grey-500 rounded active:text-blue-500 active:outline-dotted active:outline-black focus:outline-dotted focus:outline-black hover:bg-grey-50"
```
ignorant Q: Why "bg-grey-50" here but changed to "bg-black/75" below?

LZoog replied:
The opacity class changed: https://tailwindcss.com/docs/upgrade-guide#new-opacity-modifier-syntax

pdehaan replied:
And pre-apologies for casually 👀 a draft PR, _but&hellip;_
I did run through most of the items in the upgrade guide and didn't see anything else in my `git grep` testing, apart from the 1-2 comments below. ❤️

LZoog replied:
Heheh, this is basically ready except for me poking around the styles to make sure things visually look as expected so thanks for the early review. What's strange is the opacity classes don't seem to be working as described (you can see this in the FxA settings design guide on the header border) and I think it _might_ be because we have custom colors set? The old syntax still works apparently, wasn't a "breaking" change according to the doc I sent. Still looking into it.

LZoog commented on packages/fxa-settings/src/styles/ctas.scss:
```diff
@@ -48,7 +48,7 @@
 
   &:focus {
-    @apply border-transparent outline-black-dotted;
+    @apply border-transparent outline-dotted outline-black;
```
Nice catch on both of those ([bg color opacity](https://tailwindcss.com/docs/background-color#changing-the-opacity) / [text opacity](https://tailwindcss.com/docs/background-color#changing-the-opacity)). 😁

LZoog commented on packages/fxa-admin-panel/package.json:
```diff
@@ -3,15 +3,15 @@
   "scripts": {
-    "build-postcss": "postcss src/styles/tailwind.css -o src/styles/tailwind.out.css",
-    "build:client": "tsc --build ../fxa-react && tsc --build && NODE_ENV=production npm run build-postcss && SKIP_PREFLIGHT_CHECK=true PUBLIC_URL=/ INLINE_RUNTIME_CHUNK=false CI=false rescripts build",
+    "build-css": "tailwindcss -i ./src/styles/tailwind.css -o ./src/styles/tailwind.out.css",
```
See [this comment](https://github.com/mozilla/fxa/pull/13461#discussion_r908575832), not sure why it's not showing up under files changed.

LZoog commented on packages/fxa-payments-server/.storybook/styles.scss:
```diff
@@ -1,5 +0,0 @@
-// HACK: <div id="root"> is a part of the Storybook template, try to minimize
```
I removed this file because we don't have it in any other Storybooks and I'm not sure why we need it. Happy to add it back if there's a reason to.

LZoog commented:
@StaberindeZA thanks in advance for the thorough review 🙏

I mentioned this in Slack too but I noticed on the Settings page, "missing or duplicate items?" and "get Firefox on mobile or tablet" is left-aligned while right now in prod, it's center aligned. I think our `mobileLandscape` class wasn't being properly applied and I actually checked the [original Settings designs](https://www.figma.com/file/Nj2WjMOXoJBiPoWvyJ946X/FxA---Manage-Settings-(Copy)?node-id=0%3A1) which had these left-aligned except for mobile.

prod vs now
![image](https://user-images.githubusercontent.com/13018240/176711173-c636d7cb-d202-4561-8a69-c606467300eb.png)

![image](https://user-images.githubusercontent.com/13018240/176711104-2574a34e-bcbc-4bd8-968b-9a79bee6b41c.png)

StaberindeZA commented:
I tried adding a new Tailwind class to `fxa-settings` and `fxa-payments`, and not sure things behaved the way they should.

fxa-settings
- Added `bg-green-900` to class `unit-row-content` in `unit-row.scss`. The background color did not change when I refreshed the page or when I restarted fxa-settings. After restarting all services, the background was added.

fxa-payments
- Added `bg-lime-200` to TermsAndPrivacy. The class was added, but the bg color didn't change even after restart. (Looks like the class wasn't added to tailwind.out.css)

I also noticed the following in fxa-settings when I clicked the 3x3 dots in the top right hand corner.
![image](https://user-images.githubusercontent.com/10620585/176715227-ec7120fb-5282-4a18-98b3-6138e6c1daa8.png)

LZoog commented:
@StaberindeZA I fixed the CTA problem. We were importing our custom styles _before_ Tailwind's base directives so they were being overwritten. Will push that in a bit after looking into one of your other comments.

I added `bg-green-900` to `unit-row.scss` and after just a page refresh, I see the change. If you look at the pm2 process that reruns `yarn build-css`, it does include a watch of all of our `*.scss` files, so that should work. 🤔 I'll test that adding a class on a component's TSX rebuilds it too without having to restart though, I'm not sure that it does actually since we're using pm2 and not Tailwind's watch feature so I might need to tweak that.

For your `bg-lime-200` addition, did you also add that to the Tailwind config file where we extend our colors? Since we have custom colors, what's in our shared Tailwind config will override Tailwind's default color scheme.

Oof, nice catch on the bento menu. I checked that one only in Storybook and it looked good there but I see the same problem in Settings so I'll investigate.

LZoog commented:
@StaberindeZA I believe I've addressed everything except the drop down menu, I went ahead and pushed since 🤞 on getting this merged today, looking at that issue now.

pdehaan commented on packages/fxa-react/configs/tailwind.js:
```diff
@@ -6,12 +6,8 @@ const plugin = require('tailwindcss/plugin');
-    content: ['./src/**/*.tsx', './public/index.html'],
-    whitelist: [],
-  },
+  // This targets a standard CRA setup, but can be overridden as needed
+  content: ['./src/**/*.tsx', './public/index.html'],
```
Also interesting, but 1000% non-blocking for this PR&hellip; https://tailwindcss.com/blog/tailwindcss-v3-1#first-party-type-script-types

Not sure if we can just add the JSDoc-esque `@type {import('tailwindcss').Config}` here, or if it'd need to be attached to each `module.exports` statement in each tailwind.config.js file.

StaberindeZA reviewed (approved):
LGTM! Thanks for the changes and investigating everything going on with watching for changes.
